include/core/CPersistUtils.h

```diff
@@ -977,7 +977,12 @@ class CRestorerImpl<ContainerRestore> {
                         LOG_ERROR(<< "Restoration error at " << traverser.name());
                         return false;
                     }
-                    *(i++) = std::move(value);
+                    if (i == container.end()) {
+                        LOG_ERROR(<< "Too many values for size " << N
+                                  << " array during restoration");
```

Comment:
Is there a good reason not to `return false;` here?

Reply:
Since this is being backported to old branches I was worried there might be a place where, say, we restore 4 values into a size 3 array and have never noticed because the padding means it never causes a crash.

After this is merged I'll create a followup PR just for master that makes it return `false` here.  Then there will be plenty of time for any historical bugs to be revealed by the new logging and our internal testing on master before end users get the stricter behaviour.